metric_learn/itml.py

```diff
@@ -51,52 +51,65 @@ def __init__(self, gamma=1., max_iter=1000, convergence_threshold=1e-3,
     self.A0 = A0
     self.verbose = verbose
 
-  def _process_inputs(self, X, constraints, bounds):
-    self.X_ = X = check_array(X)
+  def _process_pairs(self, pairs, y, bounds):
```

Comment:
I think it would make sense that this function `_process_pairs` is shared across the class of pair metric learners. For instance, ruling the potential pairs that are identical is useful for all algorithms

Reply:
Yes I agree, I added it to the small features TODO list at the end of the main issue: https://github.com/metric-learn/metric-learn/issues/91#user-content-featurestodo